### PR TITLE
last updated on 09/28/2015

### PR DESCRIPTION
**Pull request by sindhuh**

### CODE REVIEW

**Comment by sindhuh** [comment]
please give a look.

File: www/index.html [hunk 1]
```diff
@@ -4,23 +4,23 @@
...
-
     <script src="lib/ionic/js/ionic.bundle.js"></script>
-
+    <script src="lib/ngCordova/dist/ng-cordova.js"></script>
     <script src="cordova.js"></script>
     <script src="https://cdn.firebase.com/js/client/2.2.4/firebase.js"></script>
     <script src="https://cdn.firebase.com/libs/angularfire/1.1.2/angularfire.min.js"></script>
-
+    <script src="lib/moment/min/moment.min.js"></script>
+    <script src="lib/angular-moment/angular-moment.min.js"></script>
```
Comment (nageshwara) [inline www/index.html:15]: does angular-moment.min need moment.min ?

File: www/js/controllers.js [hunk 2]
```diff
@@ -48,55 +47,94 @@ app.controller('HistoryController',function($rootScope, $scope, Auth, UserFinish
...
-    $scope.authUser = Auth.$getAuth();
-    $scope.items = UserTasks($scope.authUser.uid);
-    $scope.finishedItems = UserFinishedTasks($scope.authUser.uid);
+app.controller('TaskController', function($scope, $state, $stateParams, $ionicHistory,
+                                          UserTask, UserTasks, Auth, $cordovaDatePicker, $interval,
+                                          $ionicListDelegate) {
 
+    $scope.authUser = Auth.$getAuth();
+    $scope.isNewTask = $stateParams.taskId == 'new';
+    $scope.username = getDisplayNameFromAuth(Auth.$getAuth()).split(" ").shift();
```
Comment (nageshwara) [inline www/js/controllers.js]: are you trying to get the first name of the user ? if so, can you rename username to firstName ?

File: www/js/controllers.js [hunk 3]
```diff
@@ -48,55 +47,94 @@ app.controller('HistoryController',function($rootScope, $scope, Auth, UserFinish
...
+    var ownerId = $stateParams.ownerId;
+    if (!ownerId) {
+        ownerId = $scope.authUser.uid;
+    }
+    $scope.setTimer = function(task){
+        if(task.dueDate == 0){
```
Comment (nageshwara) [inline www/js/controllers.js:65]: you could eliminate else if you convert condition to opposite. `simply say if (task.dueDate)`. empty if blocks looks amateur.

File: www/js/controllers.js [hunk 4]
```diff
@@ -48,55 +47,94 @@ app.controller('HistoryController',function($rootScope, $scope, Auth, UserFinish
...
 
-    $scope.showEditTask = function (item) {
-        $scope.isNewTask = item == undefined;
-        $scope.task = item ? item : {} ;
-        $scope.modal.show();
+        } else {
+            $scope.dueTime = task.dueDate - (new Date().getTime());
```
Comment (nageshwara) [inline www/js/controllers.js]: rename dueTime -> dueInterval.

File: www/js/controllers.js [hunk 5]
```diff
@@ -48,55 +47,94 @@ app.controller('HistoryController',function($rootScope, $scope, Auth, UserFinish
...
         }
-        $scope.items.$add({message: task.message}).then(function(ref) {
-            console.log(">> Successfully saved a message ", ref.key(), ref.$id);
-            task.message = '';
-            $scope.closeNewTask();
-        }).catch(function(error) {
-            // TODO show error on the modal itself.
-        });
+        function successCallback(ref) {
+            $ionicHistory.goBack();
```
Comment (nageshwara) [inline www/js/controllers.js:98]: in successCallback, you need to reset the timer. I mean, you need to call the setTimer with the updated results. also, you need to cancel the previous timer if it was set.

File: www/js/controllers.js [hunk 6]
```diff
@@ -48,55 +47,94 @@ app.controller('HistoryController',function($rootScope, $scope, Auth, UserFinish
...
-    $scope.updateTask = function (item) {
-        $scope.items.$save(item).then(function(ref) {
-            if (ref.key() == item.$id) {
-                console.log(">> Successfully updated ", ref.key());
-                $scope.closeNewTask();
-                $ionicListDelegate.closeOptionButtons();
-            }
-        }).catch(function(error) {
-        });
+    $scope.dateReminder =function(task) {
```
Comment (nageshwara) [inline www/js/controllers.js]: nit: space between = and function

File: www/templates/task.html [hunk 7]
```diff
@@ -1,5 +1,33 @@
...
+                </label>
+            </div>
+        </form>
+        <div class="padding-left padding-top">
+            Created By {{username}}
+            <span am-time-ago="task.createdAt"></span>
+        </div>
+        <div class="card">
+            <div class="item item-text-wrap color">
+                <b> No due date set yet! </b>
```
Comment (nageshwara) [inline www/templates/task.html]: dont use b. use css font styles and classes.

File: www/templates/task.html [hunk 8]
```diff
@@ -1,5 +1,33 @@
...
+                <div><b>Time Remaining : {{ dueTime | amDurationFormat : 'milliseconds'}}</b></div>
```
Comment (nageshwara) [inline www/templates/task.html]: you need to show only one of "no due date set yet" and "Time remaining".

File: www/js/controllers.js (hunk 5 above)
Comment (nageshwara) [inline www/js/controllers.js:98]: actually, not in successCallback, it should be done in `timeReminder`. but why did you do it in `catch` block ?

File: www/js/controllers.js [hunk 9]
```diff
@@ -48,55 +47,94 @@ app.controller('HistoryController',function($rootScope, $scope, Auth, UserFinish
...
+                $scope.timeReminder(task, reminderDate);
+            });
     };
+    $scope.timeReminder = function(task, reminderDate) {
+            $cordovaDatePicker.show(getDateOptions('time')).then(function (date) {
+                var reminderTime = moment(date).format("HH:mm:ss");
+                task.dueDate = moment(reminderDate + " " +reminderTime).toDate().getTime();
+                task.$save().then(function(){
+                }).catch(function(){
+                    $scope.setTimer(task);
```
Comment (nageshwara) [inline www/js/controllers.js]: $scope.setTimer should get called irrespective of success/failure. (`then` function is success. `catch` function is failure).   They are similar to `try/catch` block in java. 

also, if it is a new task, you shouldnt try to save it. you should wait for the user to save it.

File: www/index.html (hunk 1 above)
Comment (sindhuh) [inline www/index.html:15]: yes. angular-moment need moment.min

File: www/js/controllers.js (hunk 3 above)
Comment (sindhuh) [inline www/js/controllers.js:65]: yeah. But I need to do something in the if block after dueTime becomes zero sorry forgot to put TODO there

File: www/templates/task.html (hunk 8 above)
Comment (sindhuh) [inline www/templates/task.html]: yeah yeah i will make it

File: www/js/controllers.js (hunk 3 above)
Comment (nageshwara) [inline www/js/controllers.js:65]: :+1:

**Pull request merged**